java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfo.java

```diff
@@ -18,73 +18,113 @@
 
 import sleeper.core.statestore.FileInfo;
 
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class S3FileInfo {
```

Comment:
Can you add some Javadoc to the class? It needs to explain what the fields mean, particularly externalReferenceCount.